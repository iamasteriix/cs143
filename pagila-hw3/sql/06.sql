/*
 * This question and the next one are inspired by the Bacon Number:
 * https://en.wikipedia.org/wiki/Six_Degrees_of_Kevin_Bacon#Bacon_numbers
 *
 * List all actors with Bacall Number 1.
 * That is, list all actors that have appeared in a film with 'RUSSELL BACALL'.
 * Do not list 'RUSSELL BACALL', since he has a Bacall Number of 0.
 */

SELECT first_name || ' ' || last_name AS "Actor Name"
FROM actor
JOIN film_actor USING (actor_id)
WHERE film_id IN (
    SELECT film_id
    FROM film_actor
    JOIN actor actor_table USING (actor_id)
    WHERE actor_table.first_name || ' ' || actor_table.last_name = 'RUSSELL BACALL'
) AND first_name || ' ' || last_name != 'RUSSELL BACALL'
GROUP BY "Actor Name"
ORDER BY "Actor Name";
