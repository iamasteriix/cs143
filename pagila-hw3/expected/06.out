      Actor Name      
----------------------
 ADAM GRANT
 ALAN DREYFUSS
 ALBERT NOLTE
 ANGELA WITHERSPOON
 ANGELINA ASTAIRE
 ANNE CRONYN
 AUDREY BAILEY
 BURT DUKAKIS
 BURT POSEY
 CAMERON STREEP
 CAMERON ZELLWEGER
 CATE HARRIS
 CATE MCQUEEN
 CHRIS BRIDGES
 CHRISTIAN AKROYD
 CHRISTOPHER BERRY
 CUBA ALLEN
 CUBA OLIVIER
 DAN HARRIS
 DAN STREEP
 DARYL WAHLBERG
 DUSTIN TAUTOU
 ED GUINESS
 ELLEN PRESLEY
 EMILY DEE
 FAY KILMER
 FAY WINSLET
 FRANCES DAY-LEWIS
 FRANCES TOMEI
 GARY PENN
 GENE MCKELLEN
 GINA DEGENERES
 GOLDIE BRODY
 GRACE MOSTEL
 GRETA KEITEL
 GRETA MALDEN
 GROUCHO DUNST
 HARVEY HOPE
 HELEN VOIGHT
 HUMPHREY GARLAND
 IAN TANDY
 JAMES PITT
 JANE JACKMAN
 JAYNE NEESON
 JAYNE NOLTE
 JAYNE SILVERSTONE
 JEFF SILVERSTONE
 JIM MOSTEL
 JOE SWANK
 JOHNNY CAGE
 JOHNNY LOLLOBRIGIDA
 JUDE CRUISE
 JUDY DEAN
 JULIA MCQUEEN
 JULIANNE DENCH
 JULIA ZELLWEGER
 KARL BERRY
 KENNETH TORN
 KEVIN BLOOM
 KIRSTEN AKROYD
 KIRSTEN PALTROW
 LAURENCE BULLOCK
 LISA MONROE
 LIZA BERGMAN
 LUCILLE TRACY
 MARY KEITEL
 MARY TANDY
 MEG HAWKE
 MENA HOPPER
 MERYL GIBSON
 MICHELLE MCCONAUGHEY
 MINNIE KILMER
 MINNIE ZELLWEGER
 MORGAN HOPKINS
 MORGAN MCDORMAND
 MORGAN WILLIAMS
 NATALIE HOPKINS
 NICK STALLONE
 OLYMPIA PFEIFFER
 PARKER GOLDBERG
 PENELOPE MONROE
 PENELOPE PINKETT
 RAY JOHANSSON
 RIP CRAWFORD
 RIP WINSLET
 RIVER DEAN
 ROCK DUKAKIS
 RUSSELL TEMPLE
 SALMA NOLTE
 SANDRA KILMER
 SANDRA PECK
 SCARLETT BENING
 SEAN GUINESS
 SEAN WILLIAMS
 SIDNEY CROWE
 SISSY SOBIESKI
 SPENCER PECK
 SUSAN DAVIS
 SYLVESTER DERN
 TIM HACKMAN
 TOM MCKELLEN
 TOM MIRANDA
 UMA WOOD
 VIVIEN BASINGER
 VIVIEN BERGEN
 WALTER TORN
 WARREN JACKMAN
 WILLIAM HACKMAN
 WILL WILSON
 WOODY JOLIE
 ZERO CAGE
(111 rows)

